monkey/agent_plugins/payloads/ransomware/config-schema.json

```
@@ -0,0 +1,37 @@
+{
+  "properties": {
+    "file_extension": {
+      "default": ".m0nk3y",
+      "description": "The file extension that the Infection Monkey will use for the encrypted file.",
+      "pattern": "^(\\.[A-Za-z0-9_]+)?$",
+      "title": "File Extension",
+      "type": "string"
+    },
+    "linux_target_dir": {
+      "default": "",
+      "description": "A path to a directory on Linux systems that contains files you will allow Infection Monkey to encrypt. If no directory is specified, no files will be encrypted.",
+      "pattern": "^$|^/|^\\$|^~",
+      "title": "Linux Target Directory",
+      "type": "string"
```

Interesting. Previously this was a nullable type. Now it is a string. Should the options be changed to make it a string also? Does it need to be optional, or do we handle an empty string appropriately?

---

I think this is ultimately nit picking, but I think `None` is preferable to an empty string. `None` means that nothing was specified, whereas an empty string could be interpreted as an invalid path. None is more explicit. Again, though, I wouldn't waste time debating this.

The bigger issue is that the code likely handles the case where this is `None/null`, so I'd advocate we put these back the way they were.

```suggestion
      "type": ["null", "string"]
```